src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -962,6 +1000,16 @@ public void stateChanged(ChangeEvent event) {
                     ((EntryEditorTab) activeTab).updateAll();
                     activateVisible();
                 }
+
+                // When the tab "Related articles" gets selected, the request to get the recommendations is started.
+                if (((JTabbedPane) event.getSource()).getSelectedComponent().getName()
+                        .equals(Localization.lang("Related articles"))) {
+                    if (relatedAcrticlesTabNeedsRecommendations) {
```

Comment:
What I want is that the EntryEditor does not know about the caching of the recommended articles. I.e. here it just calls `relatedArticlesTab.focus()` (or whatever you want to name it) and in the focus method you have the check ala 
````
if (relatedAcrticlesTabNeedsRecommendations) {
   requestRecommendations();
}
````